ta/pkcs11/src/pkcs11_token.c

```diff
@@ -309,6 +309,12 @@ uint32_t entry_ck_token_info(uint32_t ptypes, TEE_Param *params)
 
 	pad_str(info.manufacturer_id, sizeof(info.manufacturer_id));
 	pad_str(info.model, sizeof(info.model));
+
+	if (snprintf(sn, sizeof(sn), "%0*"PRIu32,
+		     PKCS11_TOKEN_SERIALNUM_SIZE, token_id) >= (int)sizeof(sn))
+		TEE_Panic(0);
```

Comment:
maybe should more strictly test `snprintf()` does not return a negative value...

Reply:
+1